routers/api/v1/repo/board.go

```diff
@@ -0,0 +1,231 @@
+// Copyright 2022 The Gogs Authors. All rights reserved.
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package repo
+
+import (
+	"net/http"
+
+	project_model "code.gitea.io/gitea/models/project"
+	"code.gitea.io/gitea/modules/context"
+	"code.gitea.io/gitea/modules/convert"
+	"code.gitea.io/gitea/modules/setting"
+	api "code.gitea.io/gitea/modules/structs"
+	"code.gitea.io/gitea/modules/web"
+)
+
+func GetProjectBoard(ctx *context.APIContext) {
+	// swagger:operation GET /projects/boards/{id} board boardGetProjectBoard
+	// ---
+	// summary: Get project board
+	// produces:
+	// - application/json
+	// parameters:
+	// - name: id
+	//   in: path
+	//   description: id of the board
+	//   type: string
+	//   required: true
+	// responses:
+	//   "200":
+	//     "$ref": "#/responses/ProjectBoard"
+	//   "403":
+	//     "$ref": "#/responses/forbidden"
+	//   "404":
+	//     "$ref": "#/responses/notFound"
+
+	board, err := project_model.GetBoard(ctx, ctx.ParamsInt64(":id"))
+	if err != nil {
+		if project_model.IsErrProjectBoardNotExist(err) {
+			ctx.Error(http.StatusNotFound, "GetProjectBoard", err)
+		} else {
+			ctx.Error(http.StatusInternalServerError, "GetProjectBoard", err)
+		}
+		return
+	}
+
+	if board.Project.RepoID != ctx.Repo.Repository.ID {
+		ctx.Error(http.StatusUnauthorized, "GetProjectBoard", "board doesn't belong to the repository")
+		return
+	}
+
+	ctx.JSON(http.StatusOK, convert.ToAPIProjectBoard(board))
+}
+
+func UpdateProjectBoard(ctx *context.APIContext) {
+	// swagger:operation PATCH /projects/boards/{id} board boardUpdateProjectBoard
+	// ---
+	// summary: Update project board
+	// produces:
+	// - application/json
+	// consumes:
+	// - application/json
+	// parameters:
+	// - name: id
+	//   in: path
+	//   description: id of the project board
+	//   type: string
+	//   required: true
+	// - name: board
+	//   in: body
+	//   required: true
+	//   schema: { "$ref": "#/definitions/UpdateProjectBoardPayload" }
+	// responses:
+	//   "200":
+	//     "$ref": "#/responses/ProjectBoard"
+	//   "403":
+	//     "$ref": "#/responses/forbidden"
+	//   "404":
+	//     "$ref": "#/responses/notFound"
+	form := web.GetForm(ctx).(*api.UpdateProjectBoardPayload)
+
+	board, err := project_model.GetBoard(ctx, ctx.ParamsInt64(":id"))
+	if err != nil {
+		ctx.Error(http.StatusNotFound, "GetProjectBoard", err)
+		return
+	}
+
+	if board.Title != form.Title {
```

Comment:
Looks like the check is unnecessary, we can just assign the value.

Reply:
And check the board's repository id.

Reply:
what error to return if the repo ids don't match. Unauthorized of BadRequest ?

Reply:
If that, `reqRepoReader(unit.TypeProjects),` will not get the right repository and will not check the right actually. We need check the permission in the function.